typescript/test/deposit-refund.test.ts

```diff
@@ -0,0 +1,166 @@
+          const refunderAddress =
+            depositRefundOfNonWitnessDepositAndWitnessOutput.refunderAddress
+
+          const rawTransactions = new Map<string, RawTransaction>()
+          rawTransactions.set(utxo.transactionHash.toString(), {
+            transactionHex: utxo.transactionHex,
+          })
+          bitcoinClient.rawTransactions = rawTransactions
+          ;({ transactionHash } = await submitDepositRefundTransaction(
+            bitcoinClient,
+            fee,
+            utxo,
+            deposit,
+            refunderAddress,
+            refunderPrivateKey
+          ))
+        })
+
+        it("should broadcast refund transaction with proper structure", async () => {
+          expect(bitcoinClient.broadcastLog.length).to.be.equal(1)
+          expect(bitcoinClient.broadcastLog[0]).to.be.eql(
+            depositRefundOfNonWitnessDepositAndWitnessOutput.expectedRefund
+              .transaction
+          )
+        })
+
+        it("should return the proper transaction hash", async () => {
+          expect(transactionHash).to.be.deep.equal(
+            depositRefundOfNonWitnessDepositAndWitnessOutput.expectedRefund
+              .transactionHash
+          )
+        })
+      })
+    })
+
+    context(
+      "when the refund transaction is requested to be non-witness",
+      () => {
+        let transactionHash: TransactionHash
+
+        beforeEach(async () => {
+          const utxo =
+            depositRefundOfWitnessDepositAndNonWitnessOutput.deposit.utxo
+          const deposit =
+            depositRefundOfWitnessDepositAndNonWitnessOutput.deposit.data
+          const refunderAddress =
+            depositRefundOfWitnessDepositAndNonWitnessOutput.refunderAddress
+
+          const rawTransactions = new Map<string, RawTransaction>()
+          rawTransactions.set(utxo.transactionHash.toString(), {
+            transactionHex: utxo.transactionHex,
+          })
+          bitcoinClient.rawTransactions = rawTransactions
+          ;({ transactionHash } = await submitDepositRefundTransaction(
+            bitcoinClient,
+            fee,
+            utxo,
+            deposit,
+            refunderAddress,
+            refunderPrivateKey
+          ))
+        })
+
+        it("should broadcast refund transaction with proper structure", async () => {
+          expect(bitcoinClient.broadcastLog.length).to.be.equal(1)
+          expect(bitcoinClient.broadcastLog[0]).to.be.eql(
+            depositRefundOfWitnessDepositAndNonWitnessOutput.expectedRefund
+              .transaction
+          )
+        })
+
+        it("should return the proper transaction hash", async () => {
+          expect(transactionHash).to.be.deep.equal(
+            depositRefundOfWitnessDepositAndNonWitnessOutput.expectedRefund
+              .transactionHash
+          )
+        })
+      }
+    )
+  })
+
+  describe("assembleDepositRefundTransaction", () => {
+    // TODO
```

Comment:
We decided to add unit tests for `assembleDepositRefundTransaction` in the follow-up PR.